packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -436,16 +444,32 @@ export class ConnectionManager implements IConnectionManager {
 
 		this._outbound.clear();
 
+		let text = "Closing DeltaManager";
+		let error: ICriticalContainerError | undefined;
+		let finalSwitchToReadonly = switchToReadonly;
+
+		// Handle overloads
+		if (
+			typeof disconnectReasonOrError === "string" &&
+			disconnectReasonOrError in DisconnectReason
+		) {
```

Comment:
Shouldn't need to check if in DisconnectReason. Typing says it is since that is the only type that is a string. You may need a helper function if typeof is not enough to keep tsc happy.